### PR TITLE
Move extra-md files to a target dir, instead of tut-sources

### LINKED ISSUE #194: micrositeExtraMdFiles copies files in to the tut dir
I think copying the `.md` files in to the `tut` sources dir should be opt-in (or there should be a way to opt-out). If they're including extra markdown sources that are not in the `tut` directory, there is probably a reason that they are not there.

### PR DESCRIPTION
**Pull request by Jacoby6000**
Resolves #194

### CODE REVIEW

**Comment by juanpedromoreno** [comment]
This is great! Travis failed though (I've just restarted it to see if the issue is gone).

Anyway, would you mind to add scripted tests to ensure this behavior.
Many thanks for your time!

**Comment by Jacoby6000** [comment]
Seems to have passed. I wonder if I should make the extra md output directory configurable. I can imagine a situation where people may want to define a location for it.  I'll add that to this PR.

**Comment by Jacoby6000** [comment]
So, I was able to figure out how to run the tests, and I added a note for running them in `CONTRIBUTING.md`. However, it seems that they do not run properly locally.  They always pull in 0.6.2-SNAPSHOT remotely, rather than using a locally published version. After deleting the local ivy cache and rerunning `publishLocal`, I get this error 

```
[error] /private/var/folders/5s/7gqq0sgd0_1dsyygvmcsspw80000gn/T/sbt_86ae426/change-default-paths/build.sbt:13: error: not found: value micrositeExtraMdFilesOutput
[error] micrositeExtraMdFilesOutput := (target in Compile).value / "extra_md_override"
[error] ^
```
Travis doesn't seem to have this problem though.

**Comment by juanpedromoreno** [comment]
Anyway, do you know you can debug a single scripted test in isolation? As it would be a standard sbt project:

For example, in my case:

```
cd /Users/juanpedromoreno/temp/sbt-microsites/src/sbt-test/microsites/change-default-paths
sbt -Dplugin.version=0.6.2-SNAPSHOT

... // and then, run all the commands in your test script, one by one
```

File: src/test/scala/microsites/util/Arbitraries.scala [hunk 1]
```diff
@@ -160,7 +161,8 @@ trait Arbitraries {
           micrositeExternalLayoutsDirectory,
           micrositeExternalIncludesDirectory,
           micrositeDataDirectory,
-          micrositeExtraMdFiles
+          micrositeExtraMdFiles,
+          micrositeExtraMdFilesOutput
```
Comment (Jacoby6000) [inline src/test/scala/microsites/util/Arbitraries.scala]: Ah. Seems to be because I missed the shift key on line 126 earlier.

**Comment by juanpedromoreno** [comment]
LGTM!

is it already ready to merge or do you want to do additional changes?

Thanks for your time @Jacoby6000 !

**Comment by Jacoby6000** [comment]
I think it's ready, I can't think of anything else I want to change.  

One thing to note is that to run tut on a specific dir, I'm directly calling a function from tut, because it seems that tut does not support calling something like `tutDir ./foo/`. If you would like, I can try to make it defer to `tutOnly` and run `tutOnly` on each file within `micrositeExtraMdFiles`, rather than aggregating all the extra md to a new location, and then directly calling a function from tut.

**Comment by juanpedromoreno** [comment]
👍 

I think the result might not be a big difference in terms of performance. I would merge it as it is, it's a great job.

However, feel free to experiment with it, in order to figure out what would be best, we'll be happy to help and accept new improvements :)

Thanks again!

**Pull request merged**